### PR TITLE
Giftcodes

### PR DESCRIPTION
**Pull request by dpaiv0**
# What will be implemented 
+ `s!giftcode <create> <value>` - creates a giftcode of x Switchcoins, while taking money from the issuer
+ `s!giftcode <list>` - lists giftcards issued by user, and their statuses
+ `s!giftcode <redeem> <code>` - redeems giftcode, and adds money from giftcode (if is valid)
+ `this.openedDms` - requirement to check if a user has their DMs opened or not (still wip, need help)

### CODE REVIEW

File: src/commands/bot/giftcode.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    const embed = new SwitchbladeEmbed(author)
+    giftcode = giftcode.toUpperCase()
+    channel.startTyping()
+    const giftcodeDocument = await this.client.database.giftcodes.findOne(giftcode)
+    if (!giftcodeDocument) {
+      embed.setColor(Constants.ERROR_COLOR)
+        .setTitle(t('commands:giftcode.subcommands.redeem.invalidGiftcode'))
+    } else if (giftcodeDocument.claimed) {
+      embed.setColor(Constants.ERROR_COLOR)
+        .setTitle(t('commands:giftcode.subcommands.redeem.alreadyRedeemed'))
```
Comment (perronosaurio) [inline src/commands/bot/giftcode.js]: wheres giftcodeDocument.claimedBy ?

Comment (bolsomito) [reply src/commands/bot/giftcode.js]: not going to be used, for privacy reasons

File: src/commands/bot/giftcode.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  }
+}
+
+class GiftcodeGenerate extends Command {
+  constructor (client, parentCommand) {
+    super(client, parentCommand)
+    this.name = 'generate'
+
+    this.parameters = new CommandParameters(this,
+      new NumberParameter({ min: 1, max: 50000, missingError: 'commands:giftcode.subcommands.generate.noValue' })
```
Comment (Doges) [inline src/commands/bot/giftcode.js]:
```suggestion
      new NumberParameter({ min: 25000, max: 50000, missingError: 'commands:giftcode.subcommands.generate.noValue' })
```

**Comment by bolsomito** [comment]
same

**Comment by Doges** [comment]
> same

tenta enviar qualquer coisa

**Comment by Doges** [comment]
fixed

**Comment by bolsomito** [comment]
LMAO

**Comment by Doges** [comment]
aprendi a mexer com o codeclimate

**Comment by Doges** [comment]
@bolsomito ve dm

**Comment by bolsomito** [comment]
RFR

**Comment by dpaiv0** [comment]
closed for now since ppl like @Doges found that money from giftcards can be duped

**Comment by Doges** [comment]
i will remake it using modules

**Pull request closed without merging**